src/listing.rs

```diff
@@ -218,17 +227,43 @@ pub fn directory_listing<S>(
         }
     }
 
-    Ok(HttpResponse::Ok()
-        .content_type("text/html; charset=utf-8")
-        .body(
-            renderer::page(
-                &title,
-                entries,
-                is_root,
-                page_parent,
-                sort_method,
-                sort_order,
-            )
-            .into_string(),
-        ))
+    if let Some(compression_method) = &download {
+        match archive::create_archive_file(&compression_method, &dir.path) {
+            Ok((filename, content)) => Ok(HttpResponse::Ok()
+                .content_type(compression_method.content_type())
+                .content_length(content.len() as u64)
+                .content_encoding(compression_method.content_encoding())
+                .header("Content-Transfer-Encoding", "binary")
+                .header(
+                    "Content-Disposition",
+                    format!("attachment; filename={:?}", filename),
+                )
+                .chunked()
+                .body(Body::Streaming(Box::new(once(Ok(content)))))),
+            Err(_) => {
+                println!(
+                    "{error} an error occured while compressing {folder}",
+                    error = Paint::red("error:").bold(),
+                    folder = dir.path.display(),
+                );
+                Ok(HttpResponse::Ok()
+                    .status(http::StatusCode::INTERNAL_SERVER_ERROR)
+                    .body(""))
```

Comment:
A helpful message here wouldn't hurt.

Reply:
Yup, that's in my to-do list :)

Reply:
I'm wondering, how precise do you want the messages to be ? Most descriptions from `std::io::Error` are not really helpful on the user-end, for instance `Is a directory` doesn't says much. Should I then go deeper than that ?

I know there's the `common_failures` crate that might help us in this case, at least for the `std::io::Error`...

Reply:
Well, they don't have to be too precise on the client side but it'd be helpful to have something more than an empty text. They should be pretty accurate on the server side of things, though.

Reply:
I don't think you ever gave this any text. It should have at least a somewhat rough message for what's wrong. Even if you just print the name of the bottom-most exception.